src/v2.rs

```diff
@@ -542,6 +662,22 @@ impl EszipV2 {
     self.npm_snapshot.take()
   }
 
+  /// Configure the hash function with which to checksum the source of the modules
+  ///
+  /// Defaults to Sha256.
+  pub fn set_source_hash_function(&mut self, hash_function: HashFunction) {
```

Comment:
Question to reviewer: do we want to expose `HashFunction`, or maybe we prefer something like `EszipV2::checksum_with_crc32(&mut self)`, `EszipV2::checksum_with_xxhash(&mut self)`, etc?